File: backend/domain/goal/goal_status.py
from __future__ import annotations

from enum import StrEnum


class GoalStatus(StrEnum):
    """
    Current lifecycle state of a learning goal.
    """

    NOT_STARTED = "not_started"

    IN_PROGRESS = "in_progress"

    COMPLETED = "completed"

    PAUSED = "paused"

    CANCELLED = "cancelled"

File: frontend/lib/domain/models/goal/goal_status.dart
enum GoalStatus {
  notStarted('not_started'),
  inProgress('in_progress'),
  completed('completed'),
  paused('paused'),
  cancelled('cancelled');

  final String value;

  const GoalStatus(this.value);

  static GoalStatus fromValue(String value) {
    return GoalStatus.values.firstWhere(
      (status) => status.value == value,
      orElse: () => GoalStatus.notStarted,
    );
  }
}
